src/Domain/Donation.php

```diff
@@ -843,11 +843,10 @@ public function getTipAmount(): string
 
     public function setTipAmount(string $tipAmount): void
     {
-        /** @var numeric-string $tipAmount */
-
+        /** @psalm-var numeric-string $tipAmount */
         if (
-            $this->paymentMethodType == PaymentMethodType::CustomerBalance &&
-            bccomp($tipAmount, '0') !== 0
+            $this->paymentMethodType === PaymentMethodType::CustomerBalance &&
```

Comment:
Agreed we should use `===` whenever possible, although I don't think in this case it could do anything different to `==`.